### ISSUE TITLE
Seventh goes downward rule is missing on "viio" chords

### ISSUE THREAD

**Issue by napulen** [ekzhang/harmony#3]
In this example, the tritone (`F` and `B`) of a `viio` moves in parallel motion upwards. The parallel motion is probably allowed because the quality of the fifth is different `diminished fifth != perfect fifth` and it is not penalized as a parallel fifth.

However, the most important thing is that the `F` should preferably resolve to `E` instead of going up to `G`. This is already handled in `V7` chords with the rule of *Seventh should resolve downward or stay in unison*, however, this does not mitigate the same problem in `viio` chords, because the `F` is the `fifth` of that triad, not the seventh. The code is looking for `chord.seventh`.

![image](https://user-images.githubusercontent.com/7258463/90659916-4221f880-e213-11ea-823c-954a32909419.png)

The example can be reproduced with this input:

```python
parser.set_defaults(
        key="C",
        chord_progression="I IV6 V6 I V vi IV IV2 viio6 I V I V6 vi65 viio6 I6 ii65 V7 I",
        durations="1 1 1 1 1 1 1 1/2 1/2 1 2 1 1 1 1 1 1 1 2",
        time_signature="3/4",
)
```

In the same example, a melodic leap of a seventh is seen in the bass (`D` down a seventh to `E`). Some melodic motions in the same voice should be penalized. Sevenths are one of them. Augmented seconds are another one. Tritones are another one. Maybe more I can't remember now. This can be a separate issue.

**Comment by ekzhang** [comment]
Ah, I agree that seems a little awkward. Thanks for the expert feedback. I think large jumps (of > 7 semitones) in the bass are penalized, but maybe by not enough?

Have to think more about the fa->mi resolution in vii->I.

**Comment by napulen** [comment]
Yeah, currently the policy is to penalize large jumps. 

In practice, jumps are tricky. An augmented second is a relatively short leap (same as a minor third) but several harmony books forbid them (has to do with melodic and harmonic minor scales). Leaping an octave (especially in the bass) is totally fine, but sevenths are strictly forbidden by several theorists (basically because they're difficult to sing, or so they say). So, these weights need some fine adjustments additionally to keeping jumps short.

I'll try to think of something too, and maybe make a PR with the suggestions.
